packages/app/src/cli/api/graphql/app_active_version.ts

```diff
@@ -0,0 +1,51 @@
+import {gql} from 'graphql-request'
+
+export const ActiveAppVersionQuery = gql`
+  query activeAppVersion($apiKey: String!) {
+    app(apiKey: $apiKey) {
+      activeAppVersion {
+        appModuleVersions {
+          registrationId
+          registrationUuid
+          registrationTitle
+          type
+          specification {
+            identifier
+            name
+            options {
+              managementExperience
+            }
+          }
+        }
+      }
+    }
+  }
+`
+
+export interface ActiveAppVersionQueryVariables {
+  apiKey: string
+}
+
+interface AppModuleVersionSpecification {
+  identifier: string
+  name: string
+  options: {
+    managementExperience: 'cli' | 'custom' | 'dashboard'
+  }
+}
+
+interface AppModuleVersion {
+  registrationId: string
+  registrationUuid: string
+  registrationTitle: string
+  type: string
+  specification: AppModuleVersionSpecification
```

Comment:
The problem with `specification` being optional is that there is no way to identify which remote registrations included in the version are `dashboard managed` or not (`allAppExtensionRegistrations` request returns them classified from the server side)
@ani-shopify @gordonhirsch why this `specification` is optional from server side?, is there another way to return if the registration is `cli` or `dashboard` managed?

Reply:
I'm guessing the intent was that the specification wasn't necessary in some places where the query was being used, but there's nothing in the domain itself where an app module wouldn't have a specification. I'd expect that if we're filtering out anything with gating requirements, then the app modules would be filtered out too. 

IMO an app module without an associated specification should result in an error, but I don't have good context in this area of the codebase.

Reply:
Ok. I will made `specification` optional in the query. Right now we are just using it to check if the `app module` is dashboard managed or not thus the diff message will put add it to a section. In case this is undefined CLI will consider it an non-dashboard.